src/index.js

```diff
@@ -65,8 +66,13 @@ export default function urlJoin(...parts) {
         url += '/';
     }
 
-    // Finally add afterPathname (?queryString)
-    url += afterPathname;
+    // If we have a query object to be handled, lets properly stringify it and append it to the url.
+    // This will override a previous query string in url
+    if (options.query) {
+        url += `?${queryString.stringify(options.query)}`;
```

Comment:
I think we should allow the user to specify how the queryString will be serialised, see: https://github.com/sindresorhus/query-string#stringifyobject-options.

Perhaps via a `queryOptions` property?

Reply:
Yes! `queryOptions` looks good.